src/topp/IDFileConverter.cpp

```diff
@@ -615,28 +617,49 @@ class TOPPIDFileConverter :
     else if (out_type == FileTypes::FASTA)
     {
       Size count = 0;
-      ofstream fasta(out.c_str(), ios::out);
+
+      bool concat = getFlag_("concatenate_peptides");
+      //To avoid creating (K/R)P sites and losing information when the FASTA entry is used by
+      //a search engine; Peptides beginning with 'P' are saved seperatly and later moved to the beginning of the concatenated sequence.
+      String all_p(""); //peptides beginning with 'P'
+      String all_but_p(""); //all the others
+
+      FASTAFile f;
+      f.writeStart(out);
+      FASTAFile::FASTAEntry entry;
       for (Size i = 0; i < peptide_identifications.size(); ++i)
       {
-        for (Size l = 0; l < peptide_identifications[i].getHits().size(); ++l)
+        if (concat)
+        {
+          if (peptide_identifications[i].getHits()[0].getSequence().toString()[0] == 'P')
+          {
+            all_p += peptide_identifications[i].getHits()[0].getSequence().toString();
+          }
+          all_but_p += peptide_identifications[i].getHits()[0].getSequence().toString();
+        }
+        else
         {
-          const PeptideHit& hit = peptide_identifications[i].getHits()[l];
-          String seq = hit.getSequence().toUnmodifiedString();
-          std::set<String> prot = hit.extractProteinAccessionsSet();
-          fasta << ">" << seq
-                << " " << ++count
-                << " " << hit.getSequence().toString()
-                << " " << ListUtils::concatenate(StringList(prot.begin(), prot.end()), ";")
-                << "\n";
-          // FASTA files should have at most 60 characters of sequence info per line
-          for (Size j = 0; j < seq.size(); j += 60)
+          for (Size l = 0; l < peptide_identifications[i].getHits().size(); ++l)
```

Comment:
writing out all hits when in non-concat mode is a bit inconsistent.
Do we actually want this?
If yes, should there be an option of how many hits to write (default=1) for both modes?

Reply:
I'm not sure if we want this. This is the way it is done currently, but I'll happily add an input option for this.

Reply:
I think it would be good to have this option. Yes, not necessarily for this project, but in general.